source/run/include/TG4RunManager.h

```diff
@@ -43,8 +44,8 @@ class TApplication;
 class TG4RunManager : public TG4Verbose
 {
  public:
-  TG4RunManager(
-    TG4RunConfiguration* configuration, int argc = 0, char** argv = 0);
+  TG4RunManager(TGeant4* geant4vmc, TG4RunConfiguration* configuration,
```

Comment:
This code is problematic. As TG4RunManager is created in TGeant4 ctor, it should not use TGeant4 in its own ctor, as TGeant4 creation has not yet finished ...
I will take a look at this and suggest a code change outside this code review.

Reply:
You are absolutely right, I haven't thought about that. Will also think about it.

Reply:
Hello @ihrivnac,
I was thinking about this again. Generally, one has to be careful about uninitialised members of `TGeant4` since everything else is done (`TVirtualMC` is fully constructed, memory is allocated...). However, it might still be safer to move caching the VMC pointer to e.g. to `TG4RunManager::Initialize()`? What do you think?